Metrics/AppMetrics.php

```diff
@@ -23,7 +23,10 @@ class AppMetrics implements PreRequestMetricsCollectorInterface, RequestMetricsC
 
     private float $startedAt = 0;
 
-    public function __construct(private readonly LabelResolver $labelResolver, private readonly ?MetricInfoResolverInterface $metricInfoResolver = null)
+    /**
+     * @param array<float> $buckets
+     */
+    public function __construct(private readonly LabelResolver $labelResolver, private readonly array $buckets, private readonly ?MetricInfoResolverInterface $metricInfoResolver = null)
```

Comment:
If we put $buckets as a last argument and make it have a default value (maybe nullable to set defaults) it will keep the signature compatible with the previous one. WDYT?

Reply:
I just broke the signature in #121. It's not yet released, so why bother now?

Reply:
I still think that buckets should go as a last argument. Subjectively, it feels to be a better order of arguments. I don't have a better explanation, sorry. Shouldn't be a big deal to change it ;)

Reply:
Thank you for your answer.

I see. Any ideas on how to deal with [optional parameters before required ones](https://www.php.net/manual/en/migration81.incompatible.php#migration81.incompatible.core.optional-before-required) in that case?

Reply:
If we do it this way

```suggestion
    public function __construct(private readonly LabelResolver $labelResolver, private readonly ?MetricInfoResolverInterface $metricInfoResolver = null, private readonly ?array $buckets = null)
```

we can handle null $buckets in the constructor if not provided.